src/main/kotlin/net/minecrell/pluginyml/paper/PaperPluginDescription.kt

```diff
@@ -56,28 +57,34 @@ class PaperPluginDescription(project: Project) : PluginDescription() {
     @Input @Optional @JsonProperty("has-open-classloader") var hasOpenClassloader: Boolean? = null
     @Input @Optional @JsonProperty("folia-supported") var foliaSupported: Boolean? = null
 
-    @Nested @Optional @JsonProperty("dependencies") @JsonSerialize(`as` = Collection::class)
-    var depends: NamedDomainObjectContainer<DependencyDefinition> = project.container(DependencyDefinition::class.java)
-    @Nested @Optional @JsonProperty("load-before") @JsonSerialize(`as` = Collection::class)
-    var loadBefore: NamedDomainObjectContainer<LoadDefinition> = project.container(LoadDefinition::class.java)
-    @Nested @Optional @JsonProperty("load-after") @JsonSerialize(`as` = Collection::class)
-    var loadAfter: NamedDomainObjectContainer<LoadDefinition> = project.container(LoadDefinition::class.java)
+    @Nested @Optional @JsonIgnore
+    var serverDependencies: NamedDomainObjectContainer<DependencyDefinition> = project.container(DependencyDefinition::class.java)
+    @Nested @Optional @JsonIgnore
+    var bootstrapDependencies: NamedDomainObjectContainer<DependencyDefinition> = project.container(DependencyDefinition::class.java)
+
+    @JsonProperty("dependencies") @JsonSerialize(contentAs = Collection::class)
```

Comment:
Are you sure this is serialized correctly? In the announcement you have in your description the format changed from a `Collection` to a map.

Instead of

```yaml
  - name: RegistryPlugin
    required: false
```

you now want

```yaml
  RegistryPlugin:
    required: false
```

To get that you need to remove the `@JsonSerialize` (and use the default). Also, the `name` of `DependencyDefinition` must be `@JsonIgnore` so it's not included in the YAML, similar to `BukkitPluginDescription.Command`.

Please check that the resulting YAML looks like you want to have it :)

Reply:
I realized this right before you posted this comment and came to the exact same solution: https://github.com/Minecrell/plugin-yml/pull/34/commits/c6eb8e213fb6c01bafa590b5caadb69b82a160eb